src/napari_imagej/widgets/menu.py

```diff
@@ -286,3 +286,66 @@ def disable_popup(self):
         msg.setTextFormat(Qt.RichText)
         msg.setTextInteractionFlags(Qt.TextBrowserInteraction)
         msg.exec()
+
+
+class SettingsButton(QPushButton):
+
+    # Signal used to identify changes to user settings
+    setting_change = Signal()
+
+    def __init__(self, viewer: Viewer):
+        super().__init__()
+        self.viewer = viewer
+
+        icon = QColoredSVGIcon(resource_path("gear"))
+        self.setIcon(icon.colored(theme=viewer.theme))
+
+        self.clicked.connect(self._update_settings)
+        self.setting_change.connect(self._notify_settings_change)
+        self.setting_change.connect(self._write_settings)
+
+    def _update_settings(self):
+        """
+        Spawns a popup allowing the user to configure napari-imagej settings.
+        """
+        args = {}
+        # Build values map by iterating over all default settings.
+        # NB grabbing settings and defaults has two benefits. Firstly, it
+        # ensures that settings ALWAYS appear in the order described in
+        # config-default.yaml. Secondly, it ensures that tampering with any user
+        # setting files does not add settings to this popup that don't exist in
+        # the default file
+        default_source = next(s for s in settings.sources if s.default)
```

Comment:
Might want to provide a default here in case we can't find our default config.

Reply:
Is there really no API for accessing the default configuration?

Reply:
There is no API for accessing the default source, but we can at least give a nicer error (in b219169) if we somehow missed it.